test/unit_tests/braket/experimental/autoqasm/test_gate_decorator.py

```diff
@@ -358,3 +358,24 @@ def subroutine(int[32] q0, int[32] q1) {
 subroutine(2, 3);"""
 
     assert main.build().to_ir() == expected
+
+
+def test_gate_modifiers() -> None:
+    @aq.gate
+    def t(q: aq.Qubit):
+        rz(q, np.pi / 4)
+
+    @aq.main(num_qubits=4)
+    def main():
+        t(1, control=0)
+        t(2, control=[0, 1], control_state="10", power=0.123)
```

Comment:
Could be interesting to think of alternatives, like
```
aq.ctrl([0, 1], "10") @ aq.pow(0.123) @ t(1)
```

Reply:
I agree that there may be a more clever syntax for this. The approach in this PR is most consistent with what the Braket SDK does today (and also happens to be the easiest to implement, given how OQpy implements control modifiers). But we should keep thinking about this and consider whether there's something nicer to do.